deployable/java/src/main/java/envtest/deployable/DeployableApplication.java

```diff
@@ -0,0 +1,157 @@
+            String line = reader.readLine();
+            return line;
+        } catch (IOException e){
+            System.out.println(e);
+        }
+        throw new RuntimeException("could not find project ID");
+    }
+
+    private static void startPubsubSubscription() throws IOException, RuntimeException {
+        // create variables
+        String projectId = getProjectId();
+        String topicId = System.getenv().getOrDefault("PUBSUB_TOPIC", "logging-test");
+        String subscriptionId = topicId + "-subscriber";
+        TopicName topicName = TopicName.of(projectId, topicId);
+        ProjectSubscriptionName subscriptionName = ProjectSubscriptionName.of(projectId, subscriptionId);
+        // create subscription
+        SubscriptionAdminClient subscriptionClient = SubscriptionAdminClient.create();
+        subscriptionClient.createSubscription(
+               subscriptionName,
+               topicName,
+               PushConfig.newBuilder().build(),
+               20);
+        // define callback
+        MessageReceiver receiver = (PubsubMessage message, AckReplyConsumer consumer) -> {
+          consumer.ack();
+          String fnName = message.getData().toStringUtf8();
+          Map<String, String> args = message.getAttributes();
+          triggerSnippet(fnName, args);
+        };
+        // start subscriber
+        Subscriber subscriber = null;
+        try {
+          subscriber = Subscriber.newBuilder(subscriptionName, receiver).build();
+          subscriber.startAsync().awaitRunning();
+          System.out.printf("Listening for messages on %s:\n", subscriptionName.toString());
+          subscriber.awaitTerminated();
+        } finally {
+          subscriber.stopAsync().awaitTerminated();
+        }
+    }
+
+    public static void triggerSnippet(String fnName, Map<String,String> args) {
+      try {
+          Snippets obj = new Snippets();
+          Class c = obj.getClass();
+          Method found = c.getDeclaredMethod(fnName, new Class[] {Map.class});
+          found.invoke(obj, args);
+      } catch (NoSuchMethodException | IllegalAccessException | InvocationTargetException e) {
+          System.out.println(e.toString());
+      }
+    }
+
+    public static void main(String[] args) throws IOException, RuntimeException {
+        String projectId = "";
+        String topicId;
+        String subscriptionId;
+
+
+        Logger root = (Logger)LoggerFactory.getLogger(org.slf4j.Logger.ROOT_LOGGER_NAME);
+        root.setLevel(Level.INFO);
+
+        // ****************** GAE, GKE, GCE ******************
+        // Enable app subscriber for all environments except GCR
+        Boolean enableSubscriber = Boolean.parseBoolean(System.getenv().getOrDefault("ENABLE_SUBSCRIBER", "false"));
+        System.out.format("ENV: ENABLE_SUBSCRIBER=true\n");
+        if (enableSubscriber) {
+          // start a pub/sub server and listen for messages
+          startPubsubSubscription();
```

Comment:
I think there is a bug here: if these run, upon completion the method ends and we continue to GCR/GAE part and happen to fail only because env var is likely not defined. 
It should be safer and more clear to "return" after line 146.